File: python-api/petstore/base_client.py
import requests

class PetstoreBaseClient:
    BASE_URL = "https://petstore.swagger.io/v2"

    def _get(self, endpoint, **kwargs):
        return requests.get(f"{self.BASE_URL}{endpoint}", **kwargs)

    def _post(self, endpoint, json=None, **kwargs):
        return requests.post(f"{self.BASE_URL}{endpoint}", json=json, **kwargs)

    def _delete(self, endpoint, **kwargs):
        return requests.delete(f"{self.BASE_URL}{endpoint}", **kwargs)

File: pw-project/api/client/baseClient.ts
import { APIRequestContext, expect } from '@playwright/test';

export class BaseClient {
  constructor(
    protected request: APIRequestContext,
    protected baseUrl: string
  ) {}

  async get(endpoint: string, options = {}) {
    const response = await this.request.get(`${this.baseUrl}${endpoint}`, options);
    expect(response.ok()).toBeTruthy();
    return response;
  }

  async post(endpoint: string, data: any, options = {}) {
    const response = await this.request.post(`${this.baseUrl}${endpoint}`, {
      data,
      ...options
    });
    expect(response.ok()).toBeTruthy();
    return response;
  }

  async put(endpoint: string, data: any, options = {}) {
    const response = await this.request.put(`${this.baseUrl}${endpoint}`, {
      data,
      ...options
    });
    expect(response.ok()).toBeTruthy();
    return response;
  }

  async delete(endpoint: string, options = {}) {
    const response = await this.request.delete(`${this.baseUrl}${endpoint}`, options);
    expect(response.ok()).toBeTruthy();
    return response;
  }
}